client/my-sites/picker/picker.jsx

```diff
@@ -92,7 +92,7 @@ class SitePicker extends Component {
 	filterSites = ( site ) => {
 		// Filter out the sites on WPCOM that don't have full Jetpack plugin installed
 		// Such sites should work fine on Jetpack Cloud
-		return getJetpackActivePlugins( site ) ? isJetpackSiteOrJetpackCloud( site ) : true;
+		return getJetpackActivePlugins( site )?.length ? isJetpackSiteOrJetpackCloud( site ) : true;
```

Comment:
Interesting!

Should we take this as an opportunity to abstract this filter function and write some tests around it?

Reply:
Thanks for pointing that out. I went a step further and refactored this code. `getJetpackActivePlugins` was used only to check for any active Jetpack plugins, so I renamed this selector to `hasJetpackActivePlugins`, fixed implementation, and added tests to check three cases.